CryptoPkg/Include/Protocol/OneCrypto.h

```diff
@@ -0,0 +1,5271 @@
+
+  @since 1.0
+  @ingroup Tls
+**/
+typedef EFI_STATUS (EFIAPI *SHARED_TLS_GET_EXPORT_KEY)(
+  IN     VOID        *Tls,
+  IN     CONST VOID  *Label,
+  IN     CONST VOID  *Context,
+  IN     UINTN       ContextLen,
+  OUT    VOID        *KeyBuffer,
+  IN     UINTN       KeyBufferLen
+  );
+
+/**
+  Verifies the validity of a RFC3161 Timestamp CounterSignature embedded in PE/COFF Authenticode
+  signature.
+
+  If AuthData is NULL, then return FALSE.
+
+  @param[in]  AuthData     Pointer to the Authenticode Signature retrieved from signed
+                           PE/COFF image to be verified.
+  @param[in]  DataSize     Size of the Authenticode Signature in bytes.
+  @param[in]  TsaCert      Pointer to a trusted/root TSA certificate encoded in DER, which
+                           is used for TSA certificate chain verification.
+  @param[in]  CertSize     Size of the trusted certificate in bytes.
+  @param[out] SigningTime  Return the time of timestamp generation time if the timestamp
+                           signature is valid.
+
+  @retval  TRUE   The specified Authenticode includes a valid RFC3161 Timestamp CounterSignature.
+  @retval  FALSE  No valid RFC3161 Timestamp CounterSignature in the specified Authenticode data.
+  
+  @since 1.0
+  @ingroup Timestamp
+**/
+typedef BOOLEAN (EFIAPI *SHARED_IMAGE_TIMESTAMP_VERIFY)(
+  IN  CONST UINT8  *AuthData,
+  IN  UINTN        DataSize,
+  IN  CONST UINT8  *TsaCert,
+  IN  UINTN        CertSize,
+  OUT EFI_TIME     *SigningTime
+  );
+
+/**
+  Gets the cryptographic provider version information.
+  
+  This function returns the version string of the cryptographic provider
+  (e.g., OpenSSL, MbedTLS, SymCrypt) that was used to compile
+  the cryptographic library.
+  
+  @retval Pointer to cryptographic provider version string.
+
+  @since 1.0
+  @ingroup Info
+**/
+typedef
+CONST CHAR8 *
+(EFIAPI *SHARED_GET_CRYPTO_PROVIDER_VERSION_TEXT)(
+  VOID
+  );
+
+/**
+  Gets the cryptographic provider version number.
+  
+  This function returns the version number of the cryptographic provider
+  (e.g., OpenSSL, MbedTLS, SymCrypt) that was used to compile
+  the cryptographic library.
+
+  @retval  Cryptographic provider version number.
+
+  @since 1.0
+  @ingroup Info
+**/
+typedef
+UINTN
+(EFIAPI *SHARED_GET_CRYPTO_PROVIDER_VERSION_NUMBER)(
+  VOID
+  );
+
+// =============================================================================
+// Protocol
+// =============================================================================
+
+/**
+  @struct _ONE_CRYPTO_PROTOCOL
+  @brief This structure defines the protocol for shared cryptographic operations.
+
+  The _ONE_CRYPTO_PROTOCOL structure provides a standardized interface for
+  cryptographic functions, enabling interoperability and consistent usage across
+  different cryptographic implementations.
+
+  Supports functions from versions:
+   - 1.0
+
+  @since 1.0
+  @ingroup OneCryptoProtocol
+**/
+typedef struct _ONE_CRYPTO_PROTOCOL

```

Comment:
Put this in a union such that we have v1 and v2 and update the OneCryptoLib to be able to use the correct structure